amethyst_animation/src/systems/control.rs

```diff
@@ -478,38 +478,40 @@ where
         let node_entity = hierarchy.nodes.get(node_index).expect(
             "Unreachable: Existence of all nodes are checked in validation of hierarchy above",
         );
-        let component = rest_states
+        if let Some(component) = rest_states
             .get(*node_entity)
             .map(RestState::state)
             .or_else(|| targets.get(*node_entity))
-            .expect(
-                "Unreachable: Existence of all nodes are checked in validation of hierarchy above",
-            );
-        let sampler_control = SamplerControl::<T> {
-            control_id: control.id,
-            channel: channel.clone(),
-            state: start_state.clone(),
-            sampler: sampler_handle.clone(),
-            end: control.end.clone(),
-            after: component.current_sample(channel, apply_data),
-            rate_multiplier: control.rate_multiplier,
-            blend_weight: 1.0,
-        };
-        let add = if let Some(ref mut set) = samplers.get_mut(*node_entity) {
-            set.add_control(sampler_control);
-            None
-        } else {
-            Some(sampler_control)
-        };
-        if let Some(sampler_control) = add {
-            let mut set = SamplerControlSet::default();
-            set.add_control(sampler_control);
-            if let Err(err) = samplers.insert(*node_entity, set) {
-                error!(
-                    "Failed creating SamplerControl for AnimationHierarchy because: {}",
-                    err
-                );
+        {
+            let sampler_control = SamplerControl::<T> {
+                control_id: control.id,
+                channel: channel.clone(),
+                state: start_state.clone(),
+                sampler: sampler_handle.clone(),
+                end: control.end.clone(),
+                after: component.current_sample(channel, apply_data),
+                rate_multiplier: control.rate_multiplier,
+                blend_weight: 1.0,
+            };
+            let add = if let Some(ref mut set) = samplers.get_mut(*node_entity) {
+                set.add_control(sampler_control);
+                None
+            } else {
+                Some(sampler_control)
+            };
+            if let Some(sampler_control) = add {
```

Comment:
Can this block be moved up so that we don't need the `add` binding anymore? Seems like NLL should take care of this now.